sdk/include/opentelemetry/sdk/metrics/aggregator/sketch_aggregator.h

```diff
@@ -0,0 +1,255 @@
+#else
+      std::terminate();
+#endif
+    }
+
+    this->values_[0] += other.values_[0];
+    this->values_[1] += other.values_[1];
+    this->checkpoint_[0] += other.checkpoint_[0];
+    this->checkpoint_[1] += other.checkpoint_[1];
+
+    auto other_iter = other.raw_.begin();
+    while (other_iter != other.raw_.end())
+    {
+      raw_[other_iter->first] += other_iter->second;
+      if (raw_.size() > max_buckets_)
+      {
+        int minidx = raw_.begin()->first, minidxval = raw_.begin()->second;
+        raw_.erase(minidx);
+        raw_[raw_.begin()->first] += minidxval;
+      }
+      other_iter++;
+    }
+
+    auto other_ckpt_iter = other.checkpoint_raw_.begin();
+    while (other_ckpt_iter != other.checkpoint_raw_.end())
+    {
+      checkpoint_raw_[other_ckpt_iter->first] += other_ckpt_iter->second;
+      if (checkpoint_raw_.size() > max_buckets_)
+      {
+        int minidx = checkpoint_raw_.begin()->first, minidxval = checkpoint_raw_.begin()->second;
+        checkpoint_raw_.erase(minidx);
+        checkpoint_raw_[checkpoint_raw_.begin()->first] += minidxval;
+      }
+      other_ckpt_iter++;
+    }
+    this->mu_.unlock();
+  }
+
+  /**
+   * Returns the checkpointed value
+   *
+   * @param none
+   * @return the value of the checkpoint
+   */
+  std::vector<T> get_checkpoint() override { return this->checkpoint_; }
+
+  /**
+   * Returns the current values
+   *
+   * @param none
+   * @return the present aggregator values
+   */
+  std::vector<T> get_values() override { return this->values_; }
+
+  /**
+   * Returns the indices (or values) stored by this sketch aggregator.
+   *
+   * @param none
+   * @return a vector of all values the aggregator is currently tracking
+   */
+  virtual std::vector<double> get_boundaries() override
+  {
+    std::vector<double> ret;
+    for (auto const &x : checkpoint_raw_)
+    {
+      ret.push_back(2 * pow(gamma, x.first) / (gamma + 1));
+    }
+    return ret;
+  }
+
+  /**
+   * Returns the error bound
+   *
+   * @param none
+   * @return the error bound specified during construction
+   */
+  virtual double get_error_bound() override { return error_bound_; }
+
+  /**
+   * Returns the count of each value tracked by this sketch aggregator.  These are returned
+   * in the same order as the indices returned by the get_boundaries function.
+   *
+   * @param none
+   * @return a vector of all counts for values tracked by the aggregator
+   */
+  virtual std::vector<int> get_counts() override
+  {
+    std::vector<int> ret;
+    for (auto const &x : checkpoint_raw_)
+    {
+      ret.push_back(x.second);
+    }
+    return ret;
+  }
+
+  virtual size_t get_max_buckets() override { return max_buckets_; }
+
+private:
+  double gamma;
+  size_t max_buckets_;
+  double error_bound_;
+  std::map<int, int> raw_;
+  std::map<int, int> checkpoint_raw_;
+};
+
+}  // namespace metrics
+}  // namespace sdk
+OPENTELEMETRY_END_NAMESPACE
```

Comment:
nit: would you add a blank line before EOF for all the files in this PR?

Reply:
fixed